src/Illuminate/Testing/ParallelRunner.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace Illuminate\Testing;
+
+use Illuminate\Contracts\Console\Kernel;
+use Illuminate\Support\Facades\Schema;
+use ParaTest\Runners\PHPUnit\Options;
+use ParaTest\Runners\PHPUnit\RunnerInterface;
+use ParaTest\Runners\PHPUnit\WrapperRunner;
+use Symfony\Component\Console\Output\ConsoleOutput;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class ParallelRunner implements RunnerInterface
+{
+    /**
+     * The application resolver callback.
+     *
+     * @var \Closure|null
+     */
+    protected static $applicationResolver;
+
+    /**
+     * The original test runner options.
+     *
+     * @var \ParaTest\Runners\PHPUnit\Options
+     */
+    protected $options;
+
+    /**
+     * The output instance.
+     *
+     * @var \Symfony\Component\Console\Output\OutputInterface
+     */
+    protected $output;
+
+    /**
+     * The original test runner.
+     *
+     * @var \ParaTest\Runners\PHPUnit\RunnerInterface
+     */
+    protected $runner;
+
+    /**
+     * Creates a new test runner instance.
+     *
+     * @param  \ParaTest\Runners\PHPUnit\Options  $options
+     * @param  \Symfony\Component\Console\Output\OutputInterface  $output
+     * @return void
+     */
+    public function __construct(Options $options, OutputInterface $output)
+    {
+        $this->options = $options;
+
+        if ($output instanceof ConsoleOutput) {
+            $output = new ParallelConsoleOutput($output);
+        }
+
+        $this->runner = new WrapperRunner($options, $output);
+    }
+
+    /**
+     * Set the application resolver callback.
+     *
+     * @param  \Closure|null  $resolver
+     * @return void
+     */
+    public static function resolveApplicationUsing($resolver)
+    {
+        static::$applicationResolver = $resolver;
+    }
+
+    /**
+     * Runs the test suite.
+     *
+     * @return void
+     */
+    public function run(): void
+    {
+        try {
+            $this->runner->run();
+        } catch (Throwable $e) {
+            throw $e;
```

Comment:
You don't need the catch here, right? I think you can use  `try { } finally {}`, if you're not doing anything with the exception itself.

```suggestion
```

Reply:
Good catch.